scripts/update-dla-format/main.ts

```diff
@@ -0,0 +1,249 @@
+    if (keys.length === 0) {
+      throw new Error(`cannot find contents of bucket ${bucket} under prefix "${prefixForm}"`);
+    }
+  } catch (error) {
+    throw new Error(`cannot find contents of bucket ${bucket} under prefix "${prefixForm}"`);
+  }
+
+  keys.forEach((key) => {
+    if (key && key.startsWith(directoryForm)) {
+      directoryIsValid = true;
+    } else if (key && key.startsWith(prefixForm)) {
+      prefixIsValid = true;
+    }
+  });
+  if (prefixIsValid) {
+    let dirname = path.dirname(prefixForm);
+    if (dirname === '.') {
+      dirname = '';
+    }
+    return manipulateTrailingSlash(dirname, true);
+  }
+  if (directoryIsValid) {
+    return directoryForm;
+  }
+  throw new Error(`Couldn't make sense of contents of bucket ${bucket} under prefix "${prefixForm}"`);
+};
+
+/**
+ * handle given targetPath, either parsing standard from prefix if targetPath is undefined
+ * or massaging targetPath into a 'Directory' if a
+ * @param targetPath given target to parse or undefined if none given
+ * @param prefix only relevant if targetPath is unspecified
+ * @returns Promise<string>
+ */
+export const parseTargetPath = async (
+  targetPath: string | undefined,
+  prefix: string
+): Promise<string> => {
+  if (targetPath !== undefined) {
+    return manipulateTrailingSlash(targetPath, true);
+  }
+  const sourceDir = await parseS3Directory(prefix);
+  const defaultTarget = `${manipulateTrailingSlash(sourceDir, false)}_updated_dla/`;
+  console.log(
+    `no targetPath given, defaulting targetPath to ${defaultTarget}`
+  );
+  return defaultTarget;
+};
+
+/**
+ * pull S3 Object from sourcePath, update it to new DLA structure and push it to targetPath
+ * noop if skip is true and an object already exists at targetPath
+ * @param bucket
+ * @param sourcePath
+ * @param targetPath
+ * @param skip skip if targetPath already exists
+ * @returns whether the logic was actually run (not skipped)
+ */
+export const updateDLAFile = async (
+  bucket: string,
+  sourcePath: string,
+  targetPath: string,
+  skip: boolean = false
+): Promise<boolean> => {
+  if (skip && await s3ObjectExists({ Bucket: bucket, Key: targetPath })) {
+    return false;
+  }
+  const dlaObject = await getJsonS3Object(bucket, sourcePath);
+  const hoisted = await hoistCumulusMessageDetails(dlaObject);
+  await putJsonS3Object(bucket, targetPath, hoisted);
+  return true;
+};
+
+/**
+ * update a batch of DLA files under prefix and push them to the targetDirectory
+ * skip files that appear to already have been processed if skip is set to true
+ * @param bucket
+ * @param targetDirectory
+ * @param prefix
+ * @param skip skip files that are already present at the target directory, default false
+ */
+export const updateDLABatch = async (
+  bucket: string,
+  targetDirectory: string,
+  prefix: string,
+  skip: boolean = false
+) => {
+  const sourceDir = await parseS3Directory(prefix);
+  const subObjects = await listS3Objects(bucket, prefix);
```

Comment:
listS3Objects can skip Folders, but it doesn't list all the files.  listS3ObjectsV2 does recursive listing.